wire-ios-sync-engine/Source/UserSession/URLActionProcessors/DeepLinkURLActionProcessor.swift

```diff
@@ -36,96 +36,129 @@ class DeepLinkURLActionProcessor: URLActionProcessor {
-                       conversation.isSelfAnActiveMember {
-                        delegate.showConversation(conversation, at: nil)
-                        delegate.completedURLAction(urlAction)
-                    } else {
-                        delegate.shouldPerformActionWithMessage(conversationName, action: urlAction) { shouldJoin in
-
-                            guard shouldJoin else {
-                                delegate.completedURLAction(urlAction)
-                                return
-                            }
-
-                            ZMConversation.join(
-                                key: key,
-                                code: code,
-                                transportSession: strongSelf.transportSession,
-                                eventProcessor: strongSelf.eventProcessor,
-                                contextProvider: strongSelf.contextProvider
-                            ) { [weak self] response in
-
-                                guard let strongSelf = self else { return }
-
-                                switch response {
-                                case .success(let conversation):
-                                    strongSelf.synchronise(conversation) { result in
-                                        DispatchQueue.main.async {
-                                            switch result {
-                                            case .success(let syncConversation):
-                                                delegate.showConversation(syncConversation, at: nil)
-                                            case .failure(let error):
-                                                delegate.failedToPerformAction(urlAction, error: error)
-                                            }
-                                        }
-                                    }
-                                case .failure(let error):
-                                    delegate.failedToPerformAction(urlAction, error: error)
-                                }
-
-                                delegate.completedURLAction(urlAction)
-                            }
-                        }
-                    }
+        case .openConversation(let id):
+            handleOpenConversation(id: id, delegate: delegate)
+
+        case .openUserProfile(let id):
+            handleOpenUserProfile(id: id, delegate: delegate)
+
+        default:
+            delegate?.completedURLAction(urlAction)
+        }
+    }
+
+    private func handleJoinConversation(key: String, code: String, urlAction: URLAction, delegate: PresentationDelegate?) {
+        ZMConversation.fetchIdAndName(
+            key: key,
+            code: code,
+            transportSession: transportSession,
+            contextProvider: contextProvider
+        ) { [weak self] response in
+            guard let strongSelf = self, let delegate = delegate else {
+                return
+            }
```

Comment:
I think you can do this instead 
```suggestion
            guard let self, let delegate else {
                return
            }
```